geoportal/c2cgeoportal_geoportal/__init__.py

```diff
@@ -207,6 +207,23 @@ def get_user_from_request(request, username=None):
                 # We know we will need the role object of the
                 # user so we use joined loading
                 request.user_ = DBSession.query(User).filter_by(username=username).first()
+            else:
+                LOG.debug(
+                    "Call OAuth verify_request with:\nurl: %s\nmethod: %s\nbody:\n%s",
+                    request.current_route_url(_query=request.GET),
+                    request.method,
+                    request.body,
+                )
+                valid, oauth2_request = oauth2.get_oauth_client().verify_request(
+                    request.current_route_url(_query=request.GET),
+                    request.method,
+                    request.body,
+                    request.headers,
+                    [],
+                )
+                LOG.info("OAuth verify_request: %s", valid)
+                if valid:
+                    request.user_ = oauth2_request.user
```

Comment:
It might be more flexible to create an authentication policy.

Reply:
+1 Thanks :-)